### PR TITLE
[charts/gateway]Us896244 dbupgrade job

### PR DESCRIPTION
**Pull request by uppoju**
**Description of the change**

Added DB upgrade job for creating OTK DB
Supported Demo DB for OTK
Add read only connection support

**Benefits**
Customers can use OTK without configuring external DB for validation purpose

**Drawbacks**

NA

**Applicable issues**

NA
  - fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[charts/gateway]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: charts/gateway/production-values.yaml [hunk 1]
```diff
@@ -528,10 +535,22 @@ otk:
...
+      # properties: {"maximumPoolSize":15, "minimumPoolSize":3}
+      # jdbcURL: jdbc:mysql://<host>:<port>/<database>
+      # jdbcDriverClass: com.mysql.jdbc.Driver
+      # Oracle database name
+      # databaseName:
 
   # Install OTK Health check bundle on gateway. Uses config map.
   # Alternatively the bundle can be loaded using config map external to helm (useExisting: true)
-  # The bundle content can be found at https://github.com/CAAPIM/apim-charts/blob/stable/charts/gateway/bundles/otk-healthcheck.bundle
+  # The bundle content can be found at https://github.com/CAAPIM/apim-charts/blob/stable/charts/gateway/bundles/otkhealthcheck.bundle
```
Comment (Gazza7205) [inline charts/gateway/production-values.yaml]: otkhealthcheck.bundle doesn't appear to exist

File: charts/gateway/values.yaml [hunk 2]
```diff
@@ -478,7 +478,9 @@ service:
 #  2. restman is enabled. Can be disabled once the install/upgrage is complete (management.restman.enabled: true)
 # Note: In dual gateway installation, restart the pods after OTK install or upgrade.
 otk:
-  enabled: false
+  enabled: true
```
Comment (Gazza7205) [inline charts/gateway/values.yaml]: should be disabled (false) by default

**Review by Gazza7205 — CHANGES_REQUESTED** [review]
There's a ref in production-values.yaml that needs to be updated and default for otk.enabled should be false in values.yaml.

Chart version should be 3.0.13

Please check the linting errors with [chart-tester](https://github.com/helm/chart-testing/tree/main) locally - if the check fails the release pipeline will too

File: charts/gateway/production-values.yaml (hunk 1 above)
Comment (uppoju) [reply charts/gateway/production-values.yaml]: Done

File: charts/gateway/values.yaml (hunk 2 above)
Comment (uppoju) [reply charts/gateway/values.yaml]: Done

**Comment by uppoju** [comment]
> There's a ref in production-values.yaml that needs to be updated and default for otk.enabled should be false in values.yaml.
> 
> Chart version should be 3.0.13
> 
> Please check the linting errors with [chart-tester](https://github.com/helm/chart-testing/tree/main) locally - if the check fails the release pipeline will too

Addressed all the comments

**Review by Gazza7205 — APPROVED** [review]
LGTM

**Pull request merged**